### ISSUE TITLE
Unknown builtin op: torchvision::deform_conv2d

### ISSUE THREAD

**Issue by Jenny0420** [pytorch/pytorch#82872]
### 🐛 Describe the bug

I have a model, in this model i have used torchvision.ops.DeformConv2D
then i traced this model with out any error.
but when i want to load this jit model in c++ liobtorch

"torch::jit::load();"

i got an error about Unknown builtin op: torchvision::deform_conv2d
my Version:
python3.8 :  torch 1.11.0-cpu
                   torchvision 0.12.0-cpu
c++ : libtorch 1.11.0-cpu

"
Unknown builtin op: torchvision::deform_conv2d.
Could not find any similar ops to torchvision::deform_conv2d. This op may not exist or may not be currently supported in TorchScript.
.......
Serialized   File "code/__torch__/torchvision/ops/deform_conv.py", line 14
    bias = self.bias
    weight = self.weight
    input = ops.torchvision.deform_conv2d(argument_1, weight, offset, mask, bias, 1, 1, 1, 1, 1, 1, 1, 1, True)
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return input
"

### Versions

my Version:
python3.8 :  torch 1.11.0-cpu
                   torchvision 0.12.0-cpu
c++ : libtorch 1.11.0-cpu

**Comment by davidberard98** [comment]
Have you included c++ torchvision? It seems like your model uses some torchvision ops that are defined outside of the core pytorch libraries. You probably need to link your program with the appropriate torchvision library in order to be able to access those torchvision ops.

**Comment by Jenny0420** [comment]
@davidberard98 
yes i have already includes the torchvision c++ 
#include <torchvision/ops/ops.h>

but when i add this
#include <torchvision/vision.h>
i got another error
"
[2/2] Linking CXX executable SMOKE_libtorch
FAILED: SMOKE_libtorch 
: && /usr/bin/c++ -g  CMakeFiles/SMOKE_libtorch.dir/main.cpp.o CMakeFiles/SMOKE_libtorch.dir/src/DataProcessing.cpp.o -o SMOKE_libtorch  -Wl,-rpath,/home/yqqian/CProject/SMOKE_libtorch/third_party/libtorch/lib:/usr/local/lib:/home/yqqian/CProject/SMOKE_libtorch/third_party/torchvision/lib  ../third_party/libtorch/lib/libc10.so  ../third_party/libtorch/lib/libkineto.a  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_img_hash.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  /usr/local/lib/libopencv_world.so.4.4.0  -lyaml-cpp  ../third_party/torchvision/lib/libtorchvision.so  -Wl,--no-as-needed,"/home/yqqian/CProject/SMOKE_libtorch/third_party/libtorch/lib/libtorch.so" -Wl,--as-needed  /usr/local/lib/libopencv_world.so.4.4.0  ../third_party/libtorch/lib/libtorch.so  -Wl,--no-as-needed,"/home/yqqian/CProject/SMOKE_libtorch/third_party/libtorch/lib/libtorch_cpu.so" -Wl,--as-needed  ../third_party/libtorch/lib/libc10.so  -lpthread  /usr/lib/x86_64-linux-gnu/libpython3.8.so && :
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::is_cuda() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::mean(c10::ArrayRef<long>, bool, c10::optional<c10::ScalarType>) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::chunk(long, long) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::reshape(c10::ArrayRef<long>) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `c10::impl::OperatorEntry::reportSignatureError(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::device() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::add(c10::Scalar, c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::relu(at::Tensor const&)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `unsigned char* at::Tensor::data_ptr<unsigned char>() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::dtype() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::GradMode::set_enabled(bool)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::adaptive_avg_pool2d(at::Tensor const&, c10::ArrayRef<long>)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::options() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `torch::autograd::_wrap_outputs(std::vector<at::Tensor, std::allocator<at::Tensor> > const&, std::unordered_set<c10::TensorImpl*, std::hash<c10::TensorImpl*>, std::equal_to<c10::TensorImpl*>, std::allocator<c10::TensorImpl*> > const&, std::unordered_set<c10::TensorImpl*, std::hash<c10::TensorImpl*>, std::equal_to<c10::TensorImpl*>, std::allocator<c10::TensorImpl*> > const&, c10::ArrayRef<at::Tensor>, std::shared_ptr<torch::autograd::Node> const&)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::GradMode::is_enabled()'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::mul(c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::dropout(at::Tensor const&, double, bool)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::add_(c10::Scalar, c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::avg_pool2d(at::Tensor const&, c10::ArrayRef<long>, c10::ArrayRef<long>, c10::ArrayRef<long>, bool, bool, c10::optional<long>)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::relu_(at::Tensor&)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::sort(long, bool) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::view_as(at::Tensor const&) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::is_complex() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::unsqueeze(at::Tensor const&, long)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::from_file(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, c10::optional<bool>, c10::optional<long>, c10::TensorOptions)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::narrow(long, long, long) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::transpose_(long, long) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::clamp_(c10::optional<c10::Scalar>, c10::optional<c10::Scalar>) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::set_data(at::Tensor const&) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::scalar_tensor(c10::Scalar, c10::TensorOptions)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::relu_() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `torch::nn::Module::zero_grad()'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::fill_(c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::zero_() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::sum(c10::ArrayRef<long>, bool, c10::optional<c10::ScalarType>) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `float* at::Tensor::data_ptr<float>() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::addmm_(at::Tensor const&, at::Tensor const&, c10::Scalar, c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::__dispatch_contiguous(c10::MemoryFormat) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `c10::Half* at::Tensor::data_ptr<c10::Half>() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `double* at::Tensor::data_ptr<double>() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::transpose(long, long) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `int* at::Tensor::data_ptr<int>() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `c10::SmallVectorBase::grow_pod(void*, unsigned long, unsigned long)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::add(at::Tensor const&, c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `torch::autograd::profiler::profilerEnabled()'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::add_(at::Tensor const&, c10::Scalar) const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::transpose(at::Tensor const&, long, long)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::batch_norm(at::Tensor const&, c10::optional<at::Tensor> const&, c10::optional<at::Tensor> const&, c10::optional<at::Tensor> const&, c10::optional<at::Tensor> const&, bool, double, double, bool)'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `long* at::Tensor::data_ptr<long>() const'
/usr/bin/ld: ../third_party/torchvision/lib/libtorchvision.so: undefined reference to `at::Tensor::sub(at::Tensor const&, c10::Scalar) const'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

"


my cmakelist have already link the torchvision libraries
"
target_link_libraries(SMOKE_libtorch PUBLIC "${TORCH_LIBRARIES}" "${OpenCV_LIBS}" -lyaml-cpp TorchVision::TorchVision)
"

**Comment by davidberard98** [comment]
`FAILED: SMOKE_libtorch`: this error seems like it might be caused by ld not being able to find libtorch build location.

`my cmakelist have already link the torchvision libraries`: to confirm this, can you try to call a torchvision op from your c++ library in the same location as `torch::jit::load();` ?

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
